concepts/pivoted/eICU-pivoted-OASIS.sql

```diff
@@ -0,0 +1,571 @@
+-- This script creates a pivoted table containing the OASIS score in eICU 
+-- ------------------------------------------------------------------
+
+-- Authors:
+-- Tristan Struja, MD, MSc, MPH (ORCID 0000-0003-0199-0184) and João Matos, MS (ORICD 0000-0002-0312-1647)
+
+-- Reference for OASIS:
+--    Johnson, Alistair EW, Andrew A. Kramer, and Gari D. Clifford.
+--    "A new severity of illness scale using a subset of acute physiology and chronic health evaluation data elements shows comparable predictive accuracy*."
+--    Critical care medicine 41, no. 7 (2013): 1711-1718.
+-- https://alistairewj.github.io/project/oasis/
+
+-- Variables used in OASIS (first 24h only):
+--  Heart rate, MAP, Temperature, Respiratory rate
+--  (sourced FROM `physionet-data.eicu_crd_derived.pivoted_vital`)
+--  GCS
+--  (sourced FROM `physionet-data.eicu_crd_derived.pivoted_vital` and `physionet-data.eicu_crd_derived.physicalexam`)
+--  Urine output 
+--  (sourced  FROM `physionet-data.eicu_crd_derived.pivoted_uo`)
+--  Pre-ICU in-hospital length of stay 
+--  (sourced FROM `physionet-data.eicu_crd.patient`)
+--  Age 
+--  (sourced FROM `physionet-data.eicu_crd.patient`)
+--  Elective surgery 
+--  (sourced FROM `physionet-data.eicu_crd.patient`, `physionet-data.eicu_crd.admissiondx` and `physionet-data.eicu_crd.apachepredvar`)
+--  Ventilation status 
+--  (sourced FROM `physionet-data.eicu_crd_derived.ventilation_events`, `physionet-data.eicu_crd_derived.debug_vent_tags`, `physionet-data.eicu_crd.apacheapsvar`, 
+--   `physionet-data.eicu_crd.apachepredvar`, and `physionet-data.eicu_crd.respiratorycare`)
+
+-- Regarding missing values:
+-- Elective stay: If there is no information on surgery in an elective stay, we assumed all cases to be -> "no elective surgery"
+-- There are a lot of missing values, especially for urine output. Hence, we have created 3 OASIS summary scores:
+-- 1) No imputation, values as is with missings. 2) Imputation by assuming a Best Case scenario. 3) Imputation by assuming a Worst Case scenario. 
+
+-- Note:
+--  The score is calculated for *all* ICU patients, with the assumption that the user will subselect appropriate patientunitstayid.
+
+DROP TABLE IF EXISTS `physionet-data.eicu_crd_derived.pivoted_OASIS` ;
+CREATE TABLE `physionet-data.eicu_crd_derived.pivoted_OASIS` AS
+
+WITH 
+
+-- Pre-ICU stay LOS -> directly convert from minutes to hours
+pre_ICU_LOS_OASIS AS (
+SELECT patientunitstayid AS pid_LOS
+  ,CASE
+      WHEN hospitaladmitoffset < (0.17*60) THEN 5
+      WHEN hospitaladmitoffset BETWEEN (0.17*60) AND (4.94*60) THEN 3
+      WHEN hospitaladmitoffset BETWEEN (4.94*60) AND (24*60) THEN 0
+      WHEN hospitaladmitoffset BETWEEN (24.01*60) AND (311.80*60) THEN 2
+      WHEN hospitaladmitoffset > (311.80*60)  THEN 1
+      ELSE NULL
+      END AS pre_ICU_LOS_OASIS
+    FROM `physionet-data.eicu_crd.patient`
+    WHERE hospitaladmitoffset IS NOT NULL
+     AND patientunitstayid IS NOT NULL
```

Comment:
I removed the where clauses. However strange, I found rows with patientunitstayid NULL though. There weren't many, and I assume they will drop out because they cannot be joined anyway?